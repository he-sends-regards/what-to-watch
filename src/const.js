export const AuthorizationStatus = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`
};

export const AppRoute = {
  ROOT: `/`,
  LOGIN: `/login`,
  MY_LIST: `/mylist`,
  ADD_REVIEW: `/films/:id/review`,
  FILM: `/films`,
  PLAYER: `/player`
};

export const APIRoute = {
  FILMS: `/films`,
  LOGIN: `/login`,
  PROMO_FILM: `/films/promo`,
  REVIEWS: `/comments`,
  FAVORITE: `/favorite`
};

export const HttpCode = {
  UNAUTHORIZED: 401
};

export const FilmRateDescriptions = {
  BAD: `Bad`,
  NORMAL: `Normal`,
  GOOD: `Good`,
  VERY_GOOD: `Very good`,
  AWESOME: `Awesome`,
  UNEXPECTED_RATE: `Unexpected rate`
};

export const FilmDescriptionTabsNames = {
  OVERVIEW: `Overview`,
  DETAILS: `Details`,
  REVIEWS: `Reviews`
};

export const ALL_GENRE = `All genres`;

export const FavoriteFlags = {
  ADD_TO_FAVORITE: 1,
  REMOVE_FROM_FAVORITE: 0
};
