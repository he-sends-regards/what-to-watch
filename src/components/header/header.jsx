import React from "react";
import PropTypes from "prop-types";
import {Link} from "react-router-dom";
import {connect} from "react-redux";
import {AuthorizationStatus, AppRoute} from "../../const";

const Header = ({authorizationStatus, authInfo}) => {
  return (
    <header className="page-header">
      <div className="logo">
        <Link to={AppRoute.ROOT} className="logo__link">
          <span className="logo__letter logo__letter--1">W</span>
          <span className="logo__letter logo__letter--2">T</span>
          <span className="logo__letter logo__letter--3">W</span>
        </Link>
      </div>

      <div className="user-block">

        {
          authorizationStatus === AuthorizationStatus.AUTH
            ? <div className="user-block__avatar">
              <Link to={AppRoute.MY_LIST}>
                <img src={authInfo.avatar_url} alt="User avatar" width="63" height="63" />
              </Link>
            </div>
            : <Link className="user-block__link" style={{textDecoration: `none`, color: `inherit`, borderRadius: 0}} to={AppRoute.LOGIN}>Sign In</Link>
        }
      </div>

    </header>
  );
};

Header.propTypes = {
  authorizationStatus: PropTypes.string.isRequired,
  authInfo: PropTypes.object.isRequired
};

const mapStateToProps = ({USER}) => ({
  authorizationStatus: USER.authorizationStatus,
  authInfo: USER.authInfo
});

export {Header};
export default connect(mapStateToProps)(Header);
