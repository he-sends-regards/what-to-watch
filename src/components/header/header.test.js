import React from "react";
import {Router} from "react-router-dom";
import renderer from "react-test-renderer";
import browserHistory from "../../browser-history";
import {AuthorizationStatus} from "../../const";
import {Header} from "./header";
import mockAuthInfo from "../../mocks/authData";

describe(`Header snapshot tests`, () => {
  it(`Should Header render correctly main page and no-auth status`, () => {
    const tree = renderer
      .create(
          <Router history={browserHistory}>
            <Header authorizationStatus={AuthorizationStatus.NO_AUTH} authInfo={mockAuthInfo} />
          </Router>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`Should Header render correctly main page and no-auth status`, () => {
    const tree = renderer
    .create(
        <Router history={browserHistory}>
          <Header authorizationStatus={AuthorizationStatus.AUTH} authInfo={mockAuthInfo} />
        </Router>
    )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
