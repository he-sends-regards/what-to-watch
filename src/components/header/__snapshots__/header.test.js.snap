// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header snapshot tests Should Header render correctly main page and no-auth status 1`] = `
<header
  className="page-header"
>
  <div
    className="logo"
  >
    <a
      className="logo__link"
      href="/"
      onClick={[Function]}
    >
      <span
        className="logo__letter logo__letter--1"
      >
        W
      </span>
      <span
        className="logo__letter logo__letter--2"
      >
        T
      </span>
      <span
        className="logo__letter logo__letter--3"
      >
        W
      </span>
    </a>
  </div>
  <div
    className="user-block"
  >
    <a
      className="user-block__link"
      href="/login"
      onClick={[Function]}
      style={
        Object {
          "borderRadius": 0,
          "color": "inherit",
          "textDecoration": "none",
        }
      }
    >
      Sign In
    </a>
  </div>
</header>
`;

exports[`Header snapshot tests Should Header render correctly main page and no-auth status 2`] = `
<header
  className="page-header"
>
  <div
    className="logo"
  >
    <a
      className="logo__link"
      href="/"
      onClick={[Function]}
    >
      <span
        className="logo__letter logo__letter--1"
      >
        W
      </span>
      <span
        className="logo__letter logo__letter--2"
      >
        T
      </span>
      <span
        className="logo__letter logo__letter--3"
      >
        W
      </span>
    </a>
  </div>
  <div
    className="user-block"
  >
    <div
      className="user-block__avatar"
    >
      <a
        href="/mylist"
        onClick={[Function]}
      >
        <img
          alt="User avatar"
          height="63"
          src="https://assets.htmlacademy.ru/intensives/javascript-3/avatar/3.jpg"
          width="63"
        />
      </a>
    </div>
  </div>
</header>
`;
